### ISSUE TITLE
Dependency on document.defaultView

### ISSUE THREAD

**Issue by ividyon** [niklasvh/html2canvas#1315]
* html2canvas version tested with: 1.0.0-alpha.4
 * Browser & version: Google Chrome 63.0.3239.84 64-bit
 * Operating system: Windows 10

Hey,

love your project! We've been using it for years, but had to switch eventually as it became really outdated and didn't print correct output at all, switching to rasterizeHTML instead.

Now that you're back, I'm checking this project out, and I have a question.

In the code we've been using with rasterizeHTML so far, we try to render *separate* duplicate document, a copy of the current page, which has some elements removed. This document isn't actually rendered in the window, meaning that `document.defaultView` is empty.

However, this defaultView is used around 50 times in the code. Do you think there is a way to make the script work without?

Basically the goal is to snapshot a separate version of the page without showing it to the user, because we want to exclude some elements. Perhaps you have a way around that?

**Comment by niklasvh** [comment]
It will need some window to work in, be it a hidden iframe (as it does currently for most of the stuff). If you could show an example of your use case on jsfiddle I could take a closer look and see whether it would be feasible to fix
